version: '3.9'

services:
  # Django (с встроенным runserver)
  backend:
    build: ./backend
    command: python manage.py runserver 0.0.0.0:8000  # Используем встроенный сервер
    volumes:
      - ./backend:/app
    ports:
      - "8000:8000"
    env_file:
      - .env
    depends_on:
      - db
      - redis

  # PostgreSQL
  db:
    image: postgres:13
    env_file:
      - .env
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER}"]
      interval: 5s
      timeout: 5s
      retries: 5

  # Redis
  redis:
    image: redis:6
    ports:
      - "6379:6379"

  # Celery Worker
  celery:
    build: ./backend
    command: celery -A myproject worker --loglevel=info
    env_file:
      - .env
    depends_on:
      - db
      - redis

  # Celery Beat
  celery_beat:
    build: ./backend
    command: celery -A myproject beat --loglevel=info
    env_file:
      - .env
    depends_on:
      - db
      - redis

volumes:
  postgres_data: